peachjam/views/documents.py

```diff
@@ -9,6 +11,48 @@
 from peachjam.registry import registry
 
 
+class RedirectResolver:
+
+    RESOLVER_MAPPINGS = {
+        "africanlii": {"country_code": "aa", "domain": "agp.africanlii.org"},
+        "ghalii": {"country_code": "gh", "domain": "new.ghalii.org"},
+        "lawlibrary": {"country_code": "za", "domain": "lawlibrary.org.za"},
+        "lesotholii": {"country_code": "ls", "domain": "new.lesotholii.org"},
+        "malawilii": {"country_code": "mw", "domain": "malawilii.org"},
+        "namiblii": {"country_code": "na", "domain": "namiblii.org"},
+        "senlii": {"country_code": "sn", "domain": "senlii.org"},
+        "seylii": {"country_code": "sc", "domain": "new.seylii.org"},
+        "sierralii": {"country_code": "sl", "domain": "new.sierralii.org"},
+        "tanzlii": {"country_code": "tz", "domain": "new.tanzlii.org"},
+        "tcilii": {"country_code": "tc", "domain": "new.tcilii.org"},
+        "ulii": {"country_code": "ug", "domain": "new.ulii.org"},
+        "zambialii": {"country_code": "zm", "domain": "new.zambialii.org"},
+        "zanzibarlii": {"place_code": "tz-znz", "domain": "new.zanzibarlii.org"},
+        "zimlii": {"country_code": "zw", "domain": "zimlii.org"},
+    }
+
+    def __init__(self):
+        mapping_code = self.RESOLVER_MAPPINGS[settings.PEACHJAM["APP_NAME"].lower()]
+        self.current_authority = mapping_code.get("place_code") or mapping_code.get(
+            "country_code"
+        )
+
+    def get_url_for_frbr_uri(self, parsed_uri):
+        country_code = parsed_uri.country
+        place_code = parsed_uri.place
+        domain = None
+        if (
+            self.current_authority != place_code
+            or self.current_authority != country_code
+        ):
+            for keys, values in self.RESOLVER_MAPPINGS.items():
```

Comment:
I think you need a test for this method, because this logic is buggy :)

Reply:
@longhotsummer This logic is kinda tricky to get right. Mostly, because of the special case of zanzibarlii and tanzlii. I want zanzibarlii to be authoritative for `/akn/tz-znz` documents, while tanzlii should be authoritative for `/akn/tz-*` documents. The code checks if the request has a locality code and uses that to check for an authoritative domain. If there is none, it falls back to the country code. What happens is when on you are on zanzibarlii and you request for `akn/tz/doc/`, (without the locality) it still thinks that zanzibarlii is authoritative for that document whereas I want it to to go to tanzlii. i want this to work right and still preserve the same behavior for `akn/aa-*/`  and `akn/za-*`. The logic should still be extendable if we were, say,  to deploy a site for `aa-eac` or `za-mp`.

Reply:
Let me add some failing tests

Reply:
tests added